alerting/src/main/kotlin/org/opensearch/alerting/InputService.kt

```diff
@@ -63,6 +73,13 @@ class InputService(
                             "period_start" to periodStart.toEpochMilli(),
                             "period_end" to periodEnd.toEpochMilli()
                         )
+
+                        // Rewrite query to consider the doc ids per given index
+                        if (chainedFindingExist(indexToDocIds)) {
+                            val updatedSourceQuery = updateInputQueryWithFindingDocIds(input.query.query(), indexToDocIds!!)
```

Comment:
why are we changing at input query
add a filter after search query is constructed

Reply:
> null check for query required?

You are right. Adding the null check.

Reply:
> why are we changing at input query add a filter after search query is constructed

Since rewrittenQuery.query() returns QueryBuilder()!  (which can be null) we must do a cast to a BoolQueryBuilder (I guess) which then later we would need to set again to a rewrittenQuery.query
I don't have any more idea how to do this in elegant way - if you can give me a hint or a code snippet how I can do, it would be good. Tnx!